operators.js

```diff
@@ -263,45 +263,18 @@ function not(ops) {
 }
 
 function and(...args) {
-  return (ops, isSpecialOps) => {
-    const rhs = args
-      .map((arg) => (typeof arg === 'function' ? arg(ops, true) : arg))
-      .filter((arg) => !!arg)
-    const res =
-      ops && ops.type && !isSpecialOps
-        ? {
-            type: 'AND',
-            data: [ops, ...rhs],
-          }
-        : rhs.length > 1
-        ? {
-            type: 'AND',
-            data: rhs,
-          }
-        : rhs[0]
-    if (ops) copyMeta(ops, res)
-    return res
-  }
+  return { type: 'AND', data: args }
 }
 
 function or(...args) {
-  return (ops, isSpecialOps) => {
-    const rhs = args
-      .map((arg) => (typeof arg === 'function' ? arg(ops, true) : arg))
-      .filter((arg) => !!arg)
-    const res =
-      ops && ops.type && !isSpecialOps
-        ? {
-            type: 'OR',
-            data: [ops, ...rhs],
-          }
-        : rhs.length > 1
-        ? {
-            type: 'OR',
-            data: rhs,
-          }
-        : rhs[0]
-    if (ops) copyMeta(ops, res)
+  return { type: 'OR', data: args }
+}
+
+function where(nextOp) {
+  return (prevOp) => {
+    if (!nextOp) return prevOp
+    const res = prevOp.type ? { type: 'AND', data: [prevOp, nextOp] } : nextOp
```

Comment:
This is actually a bit easier to read than the others. If I read this correctly it is: support null, support normal case, support where outside chain.

Reply:
Yes, overall this file had a net negative git diff. You guessed correctly, but the 3rd case is "support `where` when it's the first thing right after `fromDB(db)`".

Reply:
Yes the negative diff in the amount of code is really nice and a good sign that this is a better abstraction.